libs/utils/env.py

```diff
@@ -439,6 +441,54 @@ def _init_target(self, force = False):
             self.__modules = ['bl', 'cpufreq']
             platform = Platform(model='pixel')
 
+        # Initialize gem5 platform
+        elif self.conf['board'].upper() == 'GEM5':
+            self.nrg_model = None
+            self.__modules=['cpufreq']
+
+            system_info = self.conf['board_settings']['system']
+            simulator_info = self.conf['board_settings']['simulator']
+            simulator_args = ""
+            platform_args = ""
+
+            # Get gem5 binary arguments
+            for args in simulator_info['args']:
+                simulator_args += args + " "
+            # Required for telnet terminal
+            simulator_args += "--listener-mode=on"
```

Comment:
I think you can squash a load of lines into `simulator_args = ' '.join(simulator_info['args'] + ['--listener-mode=on'])` and somether similar for `platform_args`.